sqlalchemy-stubs/sql/elements.pyi

```diff
@@ -334,34 +334,47 @@ _G = TypeVar('_G', bound=Grouping)
 
 class Grouping(ColumnElement[_T]):
     __visit_name__: str = ...
-    element: ColumnElement[_T] = ...
+    element: ClauseElement = ...
     type: TypeEngine[_T] = ...
+    @overload
     def __init__(self, element: ColumnElement[_T]) -> None: ...
+    @overload
+    def __init__(self, element: ClauseElement) -> None: ...
```

Comment:
At runtime, what will be `_T` if `ClauseElement` is given to `__init__`?

Reply:
Could you please a comment about this to the code?

Reply:
I practice, I don't believe that `Grouping()` is directly used (it's not exposed from the main module) and is only exposed from the `self_group()` method of some subclasses of `ClauseElement`. So in the general case, those methods are going to have `Grouping` typed in their return type. For the unusual case where `Grouping()` is instantiated by the user with a ClauseElement, an error will be generated by mypy:

```python
from sqlalchemy.sql.elements import Grouping, TextClause

g = Grouping(TextClause('select * from users'))  # error: Need type annotation for 'g'
```

Reply:
I have added a comment above the signature:

```python
    # the following __init__ signature produces Grouping[None] at runtime but mypy will require an annotation
    @overload
    def __init__(self, element: ClauseElement) -> None: ...
```